modules/front-end/src/app/core/services/permissions.service.ts

```diff
@@ -53,29 +53,38 @@ export class PermissionsService {
   //   return regex.test(str);
   // }
 
-  getResourceRN(resourceType: string, resource: any) {
-    switch (resourceType) {
-      case ResourceTypeEnum.Project:
-        return `project/${resource.name}`;
-      default:
-        return `resource type ${resourceType} not supported`;
-    }
-  }
+  getProjectRN = (key: string) => `${ResourceTypeEnum.Project}/${key}`;
 
-  isGranted(rn: string, action: IamPolicyAction): boolean {
-    const matchedPermissions = this.userPermissions.filter(permission => {
+  getEnvRN = (projectKey: string, envKey: string) => `${ResourceTypeEnum.Project}/${projectKey}:${ResourceTypeEnum.Env}/${envKey}`;
```

Comment:
I recommend pass the whole project and env object to `getEnvRN` function.

Reply:
I would like to pass only the info needed